osu.Game.Rulesets.Osu/UI/OsuResumeOverlay.cs

```diff
@@ -54,9 +55,16 @@ private void load()
                         // block that press event from potentially reaching a hit circle that's behind the cursor.
                         // we cannot do this from OsuClickToResumeCursor directly since we're in a different input manager tree than the gameplay one,
                         // so we rely on a dedicated input blocking component that's implanted in there to do that for us.
-                        // note this only matters when the user didn't pause while they were holding the same key that they are resuming with.
-                        if (inputBlocker != null && !drawableOsuRuleset.AsNonNull().KeyBindingInputManager.PressedActions.Contains(action))
-                            inputBlocker.BlockNextPress = true;
+                        // note this only matters when the user is resuming with a key that was not held when they paused.
+                        if (inputBlocker != null)
+                        {
+                            bool resumeKeyWasNotHeldWhenPausing = !drawableOsuRuleset.AsNonNull().KeyBindingInputManager.PressedActions.Contains(action);
+                            bool mouseOverHittableCircle = drawableOsuRuleset!.Playfield.HitObjectContainer.AliveObjects.Any(h =>
+                                h is DrawableHitCircle && h.IsHovered && !h.IsHit && drawableOsuRuleset.Playfield.HitPolicy.CheckHittable(h, Time.Current, HitResult.None) != ClickAction.Ignore);
```

Comment:
Do we care about sliders here? Afaict this won't consider them because `AliveObjects` only contains the top-level objects.

Reply:
Good spot. I've made it care via https://github.com/ppy/osu/pull/33102/commits/a895958e2430bd46ee0818199f4f4c4f9e36ed6a but I'm not sure what to think of it. It's consistent with hit circles but maybe a touch unfair? But again I'm not sure we should be catering to people who pause at critical points of the map too hard either.

Reply:
> maybe it's fine to skip on this discrepancy?

no it's not fine, what do you even mean. the point of this PR is to *stop discrepancies between replay and gameplay*.

Reply:
I'm also not sure about this. The primary goal of the resume overlay is to avoid pause buffering, and from the looks of it, this change makes it possible to pause buffer sliders (but not hitcircles). So fixing replay discrepancies at the cost of the resume overlay not working for one hitobject type. On paper this doesn't sound like an amicable solution.

@bdach would you like a fresh set of eyes on this issue at large? I haven't really been involved in the code changes or attempts to fix yet, so maybe I can come up with something that works (low confidence).